### PR TITLE
fix(api-client): strict TS config (enable `noUncheckedIndexedAccess`)

### PR DESCRIPTION
**Pull request by marclave**
**Problem**
Currently -> we have runtime errors

**Solution**
With this PR, we use TS more : )

### CODE REVIEW

File: packages/api-client/src/components/AddressBar/AddressBarServer.vue [hunk 1]
```diff
@@ -39,11 +39,13 @@ watch([activeCollection, activeRequest], ([collection, request]) => {
   if (!collection || collection.selectedServerUid || request?.selectedServerUid)
     return
 
-  collectionMutators.edit(
-    collection.uid,
-    'selectedServerUid',
-    collection.servers[0],
-  )
+  if (collection.servers?.[0]) {
```
Comment (hwkr) [inline packages/api-client/src/components/AddressBar/AddressBarServer.vue:42]: Might be worth breaking it out into something like
```
  const firstServer = collection.servers?.[0]
  if (firstServer) {
```

File: packages/api-client/src/store/active-entities.ts [hunk 2]
```diff
@@ -55,10 +55,10 @@ export const createActiveEntitiesStore = ({
   })
 
   /** Ordered list of the active workspace's collections with drafts last */
-  const activeWorkspaceCollections = computed(() =>
+  const activeWorkspaceCollections = computed<Collection[]>(() =>
```
Comment (hwkr) [inline packages/api-client/src/store/active-entities.ts:58]: 🔥

File: packages/api-client/src/views/Cookies/CookieForm.vue [hunk 3]
```diff
@@ -17,8 +18,14 @@ const options = [
   { label: 'HttpOnly', key: 'httpOnly', placeholder: 'True/False' },
 ]
 
-const activeCookie = computed(
-  () => cookies[activeCookieId.value as string] || {},
+const activeCookie = computed<Cookie>(
+  () =>
+    cookies[activeCookieId.value as string] || {
```
Comment (hwkr) [inline packages/api-client/src/views/Cookies/CookieForm.vue:23]: This `as` might also be dangerous :(

**Comment by hanspagel** [comment]
Hans sees lots of CI errors, Hans converts to draft

**Review by YtaloAzeredo — APPROVED** [review]

**Review by Davispc10 — APPROVED** [review]

File: packages/api-client/src/views/Cookies/CookieForm.vue (hunk 3 above)
Comment (geoffgscott) [reply packages/api-client/src/views/Cookies/CookieForm.vue:23]: Should be fine as there is a fallback

**Comment by amritk** [comment]
whats the latest on this?

File: packages/api-client/src/store/collections.ts [hunk 4]
```diff
@@ -39,10 +39,13 @@ export function extendedCollectionDataFactory({
...
-      ...workspaces[workspaceUid].collections,
-      collection.uid,
-    ])
+    const workspace = workspaces[workspaceUid]
+    if (workspace) {
+      workspaceMutators.edit(workspaceUid, 'collections', [
+        ...workspace.collections,
+        collection.uid,
+      ])
+    }
```
[the remark this answers is not in the document]
Comment (hanspagel) [reply packages/api-client/src/store/collections.ts:48]: makes sense

[the remark this answers is not in the document]
Comment (amritk) [reply packages/api-client/src/store/collections.ts:48]: yep we can def start logging some warnings. I'll start sprinkling it in

**Review by amritk — APPROVED** [review]
Sick PR dood

**Pull request merged**